rust/arrow/src/array/equal/mod.rs

```diff
@@ -877,17 +903,17 @@ mod tests {
                 builder.append(false).unwrap()
             }
         }
-        builder.finish().data()
+        builder.finish().data().clone()
```

Comment:
it probably doesn't matter as this is in a test, but is this `clone()` necessary? Or maybe there is just no good way to destructure the Array to give back the `ArrayData`

Reply:
Exactly, there is no way atm. This is solved with the broader proposal, for now I just cloned it. xD